File: src/Modules/Appointments/ErpClink.Modules.Appointments.Application/Appointments/Models/AppointmentModels.cs
namespace ErpClink.Modules.Appointments.Application.Appointments.Models;

public sealed record BookAppointmentRequest(
    Guid PatientId,
    Guid DoctorId,
    Guid ClinicId,
    DateOnly Date,
    TimeOnly StartTime,
    string? Reason,
    string? Notes);

public sealed record CancelAppointmentRequest(string? Reason);

public sealed record RescheduleAppointmentRequest(
    DateOnly Date,
    TimeOnly StartTime);

public sealed record SearchAppointmentsRequest(
    Guid? PatientId,
    Guid? DoctorId,
    Guid? ClinicId,
    DateOnly? Date,
    DateOnly? FromDate,
    DateOnly? ToDate,
    string? Status,
    string? AppointmentNumber,
    string? SortBy,
    bool SortDescending = false,
    int Page = 1,
    int PageSize = 20);

public sealed record AppointmentDto(
    Guid Id,
    Guid OrganizationId,
    Guid BranchId,
    string AppointmentNumber,
    Guid PatientId,
    Guid DoctorId,
    Guid ClinicId,
    DateOnly AppointmentDate,
    TimeOnly StartTime,
    TimeOnly EndTime,
    string Status,
    string? Reason,
    string? Notes,
    string? CancellationReason,
    DateTime? CancelledAtUtc,
    string? CancelledBy,
    DateTime CreatedAtUtc,
    string? CreatedBy,
    DateTime? UpdatedAtUtc,
    string? UpdatedBy);

public sealed record AppointmentListItemDto(
    Guid Id,
    string AppointmentNumber,
    Guid PatientId,
    Guid DoctorId,
    Guid ClinicId,
    DateOnly AppointmentDate,
    TimeOnly StartTime,
    TimeOnly EndTime,
    string Status);

public sealed record PagedAppointmentsResult(
    IReadOnlyList<AppointmentListItemDto> Items,
    int Page,
    int PageSize,
    int TotalCount);

File: frontend/erpclink-web/src/app/features/appointments/models/appointment.models.ts
export type AppointmentStatus =
  | 'Scheduled'
  | 'Confirmed'
  | 'Cancelled'
  | 'NoShow'
  | 'CheckedIn';

export interface BookAppointmentRequest {
  patientId: string;
  doctorId: string;
  clinicId: string;
  date: string;
  startTime: string;
  reason: string | null;
  notes: string | null;
}

export interface CancelAppointmentRequest {
  reason: string | null;
}

export interface RescheduleAppointmentRequest {
  date: string;
  startTime: string;
}

export interface SearchAppointmentsParams {
  patientId?: string;
  doctorId?: string;
  clinicId?: string;
  date?: string;
  fromDate?: string;
  toDate?: string;
  status?: string;
  appointmentNumber?: string;
  sortBy?: string;
  sortDescending?: boolean;
  page?: number;
  pageSize?: number;
}

export interface Appointment {
  id: string;
  organizationId: string;
  branchId: string;
  appointmentNumber: string;
  patientId: string;
  doctorId: string;
  clinicId: string;
  appointmentDate: string;
  startTime: string;
  endTime: string;
  status: AppointmentStatus;
  reason: string | null;
  notes: string | null;
  cancellationReason: string | null;
  cancelledAtUtc: string | null;
  cancelledBy: string | null;
  createdAtUtc: string;
  createdBy: string | null;
  updatedAtUtc: string | null;
  updatedBy: string | null;
}

export interface AppointmentListItem {
  id: string;
  appointmentNumber: string;
  patientId: string;
  doctorId: string;
  clinicId: string;
  appointmentDate: string;
  startTime: string;
  endTime: string;
  status: AppointmentStatus;
}

export interface PagedAppointmentsResult {
  items: AppointmentListItem[];
  page: number;
  pageSize: number;
  totalCount: number;
}
